assets/js/socket.js

```diff
@@ -9,16 +9,91 @@ const presence = new Presence(channel)
 
 presence.onSync(() => updateUsers(presence))
 
+let driving = false
+
 if (window.pointingParty.username) {
-  channel.join ()
+  channel.join()
     .receive('ok', resp => { console.log('Joined successfully', resp) })
     .receive('error', resp => { console.log('Unable to join', resp) })
 }
 
-const calculateButton = document.querySelector('.calculate-points')
-calculateButton.addEventListener('click', event => {
-  const storyPoints = document.querySelector('.story-points')
-  channel.push('user_estimated', { points: storyPoints.value })
+const renderTemplate = function(parent, template) {
+  while (parent.firstChild) {
+    parent.removeChild(parent.firstChild)
+  }
+
+  const fragment = document.createRange().createContextualFragment(template)
+  parent.appendChild(fragment)
+}
+
+const startButton = document.querySelector('.start-button')
+startButton.addEventListener('click', e => {
+  driving = true;
+  channel.push('start_pointing', {})
+})
+
+const cardContainer = document.querySelector('.card-container')
+channel.on('new_card', state => {
+  const template =
+    '<div class="card text-left">' +
+    '  <div class="card-header">' +
+    '    <h2>' + state.card.title + '</h2>' +
+    '  </div>' +
+    '  <div class="card-body">' +
+    '    <p class="card-text">' + state.card.description + '</p>' +
+    '    <div class="form-group text-left points-container">' +
+    '      <div class="form-row align-items-center">' +
+    '        <div class="col-2">' +
+    '          <label for="story-points">Story Points</label>' +
+    '          <select class="form-control story-points" id="story-points">' +
+    '            <option>1</option>' +
```

Comment:
I put the points in here statically for now. I didn't think making the dynamic was worth the effort. When we use LiveView I think it might make more sense?

Reply:
Yea i think when we replace with with an LEEX template we can make dynamic. I'm good with this here for this iteration of the feature.